cmd/codegen/main.go

```diff
@@ -112,18 +116,43 @@ func ClientGen(cmd *cobra.Command, args []string) error {
 		cfg.IntrospectionJSON = string(introspectionJSON)
 	}
 
-	if dependenciesJSONFilePath != "" {
-		dependenciesJSON, err := os.ReadFile(dependenciesJSONFilePath)
+	if moduleSourceID != "" {
+		dag, err := dagger.Connect(ctx)
 		if err != nil {
-			return fmt.Errorf("read dependencies json: %w", err)
+			return fmt.Errorf("failed to connect to engine: %w", err)
 		}
+		defer dag.Close()
 
-		var gitDependencies []modules.ModuleConfigDependency
-		if err := json.Unmarshal(dependenciesJSON, &gitDependencies); err != nil {
-			return fmt.Errorf("failed to unmarshal dependencies json: %w", err)
+		var res struct {
+			Source struct {
+				Dependencies []generator.ModuleSourceDependencies
+			}
+		}
+
+		err = dag.Do(ctx,
+			&dagger.Request{
+				Query:  loadModuleSourceDepsQuery,
+				OpName: "ModuleSourceDependencies",
+				Variables: map[string]any{
+					"source": dagger.ModuleSourceID(moduleSourceID),
+				},
+			},
+			&dagger.Response{
+				Data: &res,
+			})
+		if err != nil {
+			return fmt.Errorf("failed to load module source dependencies: %w", err)
+		}
+
+		cfg.ModuleDependencies = res.Source.Dependencies
+
+		// If the module source is implementing functions, we need to load the module itself
+		src, err := dag.LoadModuleSourceFromID(dagger.ModuleSourceID(moduleSourceID)).SDK().Source(ctx)
```

Comment:
Implementation wise, this gets the source of the SDK, that doesn't seem right to me.

Reply:
How could I check if the module I generate for implements functions then? If it does, I need to serve it, otherwise I shouldn't

Reply:
Oh, I see what you mean :thinking: the comment is confusing because a module can technically have an SDK and not implement any functions, this isn't what we're testing for.

Why couldn't we try and `Serve` an SDK-less module? It works :thinking: (and also means we need to handle fewer corner cases in each SDK)

Reply:
I'll check and see if it can work

Reply:
It works! That's so nice!